version_test.go

```diff
@@ -234,6 +234,8 @@ func TestComparePreReleases(t *testing.T) {
 		{"1.0.0-rc.1", "1.0.0", -1},
 		//{"1.0.0-rc9", "1.0.0-rc10", -1}, // want to support one day
 		//{"v1.0.0-rc9", "1.0.0-rc10", -1}, // want to support one day
+		{"0.9.9-r0", "0.9.12-r1", -1}, // regression
```

Comment:
regression?

Reply:
this is a regression case, that is, a specific case which was found to have caused a problem and is here to ensure we don't regress to previous bad behavior.